lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -216,6 +233,7 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
       opts
     )
 
+    send(self(), :memory_usage)
```

Comment:
This will invoke `handle_info(:memory_usage, ...)`, which will schedule a new interval. So for each evaluation, a new interval would be scheduled. I think instead you should extract the body of `handle_info(:memory_usage, ...)` into something like `send_memory_usage(state)` instead and reuse that piece from here. :)

Reply:
But isn't that the idea? If I evaluate something, I update the memory immediately and reschedule the next automatic update to after 15s. I don't think I understood the point here.

Reply:
The issue is that every time you invoke `send(self(), :memory_usage), you are setting up a new "schedule", since it calls `schedule_memory_usage` at the end. That's what you want for the first time, but you are also setting up new schedules on every evaluation.

Reply:
`Evaluator.evaluate_code` is asynchronous, so we we need to send a message from `Evaluator` to the `RuntimeServer` when it finishes evaluation, and on receiving that message we compute memory and send the report.

Reply:
We could add an option to `Evaluator.evaluate_code`, like `:notify_to` and send a message `{:evaluation_finished, ref}`.